test/jdk/java/lang/invoke/indify/Indify.java

```diff
@@ -356,149 +464,165 @@ protected Class<?> findClass(String name) throws ClassNotFoundException {
+                                System.err.println(">> Removing instruction invokestatic for Method: " + invokeInstruction.name());
+                            }
+                            else if (
+                                    shouldProceedAfterIndyAdded.peek() &&
+                                    e instanceof InvokeInstruction invokeInstruction &&
+                                    invokeInstruction.method().name().stringValue().equals("invokeExact"))
+                            {
+                                System.err.println(">> Removing instruction invokevirtual for Method: " + invokeInstruction.method());
+                                System.err.println(">> Adding invokedynamic instruction and nop instead of invoke virtual: " + ((InvokeDynamicEntry) newConstant).name());
+                                b.invokeDynamicInstruction((InvokeDynamicEntry) newConstant).nop();
+
+                                shouldProceedAfterIndyAdded.pop();
+                                shouldProceedAfterIndyAdded.push(false);
+                            } else {
+                                b.with(e);
+                            }
+                        };
+                        classTransform = ClassTransform.transformingMethodBodies(filter, codeTransform);
+
+                        classModel = of().parse(
+                               of().transform(classModel, classTransform)
+                        );
+
+                        System.out.println();
                     } else {
-                        if (!quiet)  System.err.println(i+" "+conm+" => ldc "+con);
-                        assert(i.len == 3);
-                        i.u1AtPut(0, opc_ldc_w);
-                        i.u2AtPut(1, con.index);
+                        assert(i.sizeInBytes() == 3);
+                        MethodModel finalConm = patternMethod;
+                        codeTransform = (b, e) ->{
+                            String a1 = null, a2 = null;
+                            if(e instanceof InvokeInstruction invokeInstruction){
+                                a1 = invokeInstruction.method().name().stringValue();
+                                a2 = finalConm.methodName().stringValue();
+                            }
+                            if(e instanceof InvokeInstruction invokeInstruction && Objects.equals(a1, a2)){
+                                System.err.println(":::Transfmoring the Method: "+ m.methodName() +" instruction: invokestatic " + invokeInstruction.type() + " to ldc: " +  newConstant.index() );
+                                b.ldc((LoadableConstantEntry) newConstant);
+                            } else b.with(e);
+                        };
+                        classTransform = ClassTransform.transformingMethodBodies(filter, codeTransform);
+                        classModel = of().parse(
+                             of().transform(classModel, classTransform));
```

Comment:
Injected transform to avoid verify error, that seems to me conceptually wrong.
What is the root cause of the verify error and how the transform suppose to fix it?